src/main/java/org/idpass/lite/IDPassReaderComponent.java

```diff
@@ -0,0 +1,173 @@
+                .build();
+
+        List<String> addrLines = new ArrayList<>();
+        if (idfc.getAddressLine1() != null) {
+            addrLines.add(idfc.getAddressLine1());
+        }
+        if (idfc.getAddressLine2() != null) {
+            addrLines.add(idfc.getAddressLine2());
+        }
+        if (idfc.getAddressLine3() != null) {
+            addrLines.add(idfc.getAddressLine3());
+        }
+
+        PostalAddress.Builder postalAddressBuilder = PostalAddress.newBuilder()
+                .setLanguageCode("en") /// TODO
+                .addAllAddressLines(addrLines);
+
+        if (idfc.getRegion() != null) {
+            postalAddressBuilder.setRegionCode(idfc.getRegion());
+        }
+
+        if (idfc.getProvince() != null) {
+           postalAddressBuilder.setAdministrativeArea(idfc.getProvince());
+        }
+
+        if (idfc.getPostalCode() != null) {
+            postalAddressBuilder.setPostalCode(idfc.getPostalCode());
+        }
+
+        PostalAddress postalAddress = postalAddressBuilder.build();
+
+        if (idfc.getUIN() != null) {
+            identBuilder.setUIN(idfc.getUIN());
+        }
+        if (idfc.getFullName() != null) {
+            identBuilder.setFullName(idfc.getFullName());
+        }
+
+        identBuilder.setPostalAddress(postalAddress);
+        identBuilder.setGender(idfc.getGender());
+
+        if (idfc.getGivenName() != null) {
+            identBuilder.setGivenName(idfc.getGivenName());
+        }
+        if (idfc.getSurName() != null) {
+            identBuilder.setSurName(idfc.getSurName());
+        }
+        if (idfc.getPlaceOfBirth() != null) {
+            identBuilder.setPlaceOfBirth(idfc.getPlaceOfBirth());
+        }
+
+        identBuilder.setDateOfBirth(dobProto);
+
+        Ident ident = identBuilder.build();
+        byte[] qrcodeId = null;
+
+        try {
+            Card card = reader.newCard(ident, null);
+            BufferedImage bi = card.asQRCode();
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            ImageIO.write(bi, "png", bos);
+            qrcodeId = bos.toByteArray();
+
+        } catch (IOException | IDPassException e) {
+            e.printStackTrace();
+        }
+
+        return  qrcodeId;
+    }
+
+    // Notes: copied from 'mosip-functional-tests' repo
+    private static byte[] convertJ2KToJPG(byte[] jp2Data) {
+        byte[] jpgImg = null;
+        J2KImageReader j2kImageReader = new J2KImageReader(null);
+        try {
+            j2kImageReader.setInput(ImageIO.createImageInputStream(new ByteArrayInputStream(jp2Data)));
+            ImageReadParam imageReadParam = j2kImageReader.getDefaultReadParam();
+            BufferedImage image = j2kImageReader.read(0, imageReadParam);
+            ByteArrayOutputStream imgBytes = new ByteArrayOutputStream();
+            ImageIO.write(image, "JPG", imgBytes);
+            jpgImg = imgBytes.toByteArray();
+        } catch (IOException e) {
+
```

Comment:
Should we log exception here?

Reply:
Ok. I think returning `null` instead as the caller is checking for null return value.